components/buildless-serverless/internal/controller/resources/deployment.go

```diff
@@ -126,6 +130,13 @@ func (d *Deployment) podSpec() corev1.PodSpec {
 					PeriodSeconds:    5,
 					TimeoutSeconds:   4,
 				},
+				SecurityContext: &corev1.SecurityContext{
+					RunAsGroup: d.userGroup(),
+					RunAsUser:  d.userGroup(),
```

Comment:
Do we always have the same value? are those users and group always coupled?

Reply:
No, but they are usually the same if the user and group were create together (e.g regular users also have their own groups by default)
In Python312/Nodejs20/22 we don't use any UID above 406 (except 65534); for group ID we'd need to exclude 999 as well. 
That means in our usecase we can simply use USER 1000, as in other our components.
This still will be a magic number, since we have to choose something anyway, but I think that 1000 is the most popular and standard choice.

Reply:
I'd check if this setting is needed, what are the defaults wen it's not set?